packages/@aws-cdk/aws-eks/lib/cluster-resource.ts

```diff
@@ -60,7 +60,7 @@ export class ClusterResource extends CoreConstruct {
       throw new Error('"roleArn" is required');
     }
 
-    this.clusterCreationRole = this.createAdminRole(props);
+    this.clusterCreationRole = props.clusterCreationRole;
```

Comment:
* Revert the changes to ClusterResource for the default `clusterCreationRole` when no role is supplied.

* Add conditional logic for `CreationRole`

```
if (props.eksRoleTemplateUrl) {
  this.role = iam.Role.fromRoleArn(this, 'Role', eksiam.eksServiceRoleArn);
} else {
  this.role = props.role || new iam.Role(this, 'Role', {
      assumedBy: new iam.ServicePrincipal('eks.amazonaws.com'),
      managedPolicies: [
        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSClusterPolicy'),
      ],
  });
}
```

Also, keep the line `this.clusterCreationRole = resource.clusterCreationRole;` in Cluster.

Reply:
Jeremy and I resolved all of this on screenshare.